pkg/controller/certificates/requestmanager/requestmanager_controller.go

```diff
@@ -139,17 +139,12 @@ func (c *controller) ProcessItem(ctx context.Context, key types.NamespacedName)
 	namespace, name := key.Namespace, key.Name
 
 	crt, err := c.certificateLister.Certificates(namespace).Get(name)
-	if apierrors.IsNotFound(err) {
-		log.V(logf.DebugLevel).Info("certificate not found for key", "error", err.Error())
-		return nil
-	}
-	if err != nil {
+	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
-
-	// If the Certificate object is being deleted, we don't want to create any
-	// new CertificateRequests objects
-	if crt.DeletionTimestamp != nil {
+	if crt == nil || crt.DeletionTimestamp != nil {
+		// If the Certificate object is being deleted, we don't want to create any
```

Comment:
```suggestion
		// If the Certificate object was/ is being deleted, we don't want to create any
```

Nit: just to make it consistent with the other comments.